### PR TITLE
fix: dialing peers

### PR DESCRIPTION
**Pull request by arminsabouri**

### CODE REVIEW

**Review by scottmillner — DISMISSED** [review]
🔥 🔥 🔥

**Comment by nagarajm87** [comment]
need to resolve the conflicts :)

File: crates/consensus/common/src/utils.rs [hunk 1]
```diff
@@ -65,7 +65,10 @@ pub fn is_inturn(
 
     // Determine the current signer index based on the position in the cycle
     // Each signer's turn lasts for `block_time` seconds
-    (position_in_cycle / time_range_u256) % authorities_len_u256 == signer_index_u256
+    // (position_in_cycle / time_range_u256) % authorities_len_u256 == signer_index_u256
```
Comment (rwlockbg) [inline crates/consensus/common/src/utils.rs:68]: Why was this commented out ?. The code I added does a precise calculation of whos in turn based on secs.

File: crates/net/network/src/frost/manager.rs [hunk 2]
```diff
@@ -142,6 +134,7 @@ impl FrostManager {
         data.peer_confirmed && !data.peer_commands_tx.is_closed()
     }
 
+<<<<<<< HEAD
```
Comment (rwlockbg) [inline crates/net/network/src/frost/manager.rs]: Why do we have this ? Has someone forgotten to resolve a conflict ?

File: crates/net/network/src/frost/manager.rs [hunk 3]
```diff
@@ -154,7 +147,51 @@ impl FrostManager {
...
+                    .peer_commands_tx
+                    .send(FrostPeerCommand::PeerMessage(PeerMessageResponse::Healthcheck(resp)))
+                {
+                    Ok(_) => {
+                        debug!("Healthcheck sent to peer {:?}", peer_id,);
+                    }
+                    Err(e) => {
+                        error!("Failed to send healthcheck to peer {:?}, error: {:?}", peer_id, e);
+                    }
+>>>>>>> 3cb6eb968 (fix(net|frost): keep a vec of peer connections and prune them on demand)
```
Comment (rwlockbg) [inline crates/net/network/src/frost/manager.rs]: again, conflicts need to be resolved!

File: crates/net/network/src/frost/manager.rs [hunk 4]
```diff
@@ -287,8 +324,15 @@ impl FrostManager {
                 let peer_connections = self
                     .peers_connections
                     .iter()
-                    .filter(|(_, peer_data)| Self::filter_connected_peer(peer_data))
-                    .map(|(peer_id, peer_data)| (*peer_id, peer_data.clone()))
+                    .filter(|(_, peer_data)| {
+                        // TODO this check might be redundant now that we prune closed connections
+                        Self::filter_connected_peer(
```
Comment (rwlockbg) [inline crates/net/network/src/frost/manager.rs]: I dont think you need this provided you are doing the same above in the prune function.

File: crates/consensus/common/src/utils.rs (hunk 1 above)
Comment (arminsabouri) [reply crates/consensus/common/src/utils.rs:68]: We are moving to round robin selection based on block height not time. 
For now the first fed memeber will always be the coordinator. if this is not commented here it should be

File: crates/net/network/src/frost/manager.rs (hunk 4 above)
Comment (arminsabouri) [reply crates/net/network/src/frost/manager.rs]: yes agreed. Will remove

Comment (arminsabouri) [reply crates/net/network/src/frost/manager.rs]: removed, closing

File: crates/net/network/src/frost/manager.rs (hunk 3 above)
Comment (arminsabouri) [reply crates/net/network/src/frost/manager.rs]: done

File: crates/net/network/src/frost/manager.rs (hunk 2 above)
Comment (arminsabouri) [reply crates/net/network/src/frost/manager.rs]: removed. closing

File: crates/consensus/common/src/utils.rs (hunk 1 above)
Comment (rwlockbg) [reply crates/consensus/common/src/utils.rs:68]: ok

**Review by rwlockbg — APPROVED** [review]

**Comment by arminsabouri** [comment]
https://github.com/botanix-labs/botanix/issues/768

**Review by scottmillner — APPROVED** [review]

**Pull request merged**